test/test_commands_serve.rb

```diff
@@ -12,6 +15,125 @@ def custom_opts(what)
     )
   end
 
+  def start_server(opts)
+    @thread = Thread.new do
+      merc = nil
+      cmd = Jekyll::Commands::Serve
+      Mercenary.program(:jekyll) do |p|
+        merc = cmd.init_with_program(p)
+      end
+      merc.execute(:serve, opts)
+    end
+    @thread.abort_on_exception = true
+
+    Jekyll::Commands::Serve.mutex.synchronize do
+      unless Jekyll::Commands::Serve.running?
+        Jekyll::Commands::Serve.run_cond.wait(Jekyll::Commands::Serve.mutex)
+      end
+    end
+  end
+
+  def serve(opts)
+    allow(Jekyll).to receive(:configuration).and_return(opts)
+    allow(Jekyll::Commands::Build).to receive(:process)
+
+    start_server(opts)
+
+    opts
+  end
+
+  context "using LiveReload" do
+    setup do
+      @temp_dir = Dir.mktmpdir("jekyll_livereload_test")
+      @destination = File.join(@temp_dir, "_site")
+      Dir.mkdir(@destination) || flunk("Could not make directory #{@destination}")
+      @client = HTTPClient.new
+      @client.connect_timeout = 5
+      @standard_options = {
+        "port"            => 4000,
+        "host"            => "localhost",
+        "baseurl"         => "",
+        "detach"          => false,
+        "livereload"      => true,
+        "source"          => @temp_dir,
+        "destination"     => @destination,
+        "livereload_port" => Jekyll::Commands::Serve.singleton_class::LIVERELOAD_PORT,
```

Comment:
If we have a default built in already, is there a reason to specify it here in the test?